rust/lance-file/src/v2/reader.rs

```diff
@@ -1056,4 +1061,67 @@ mod tests {
                 .is_err()
         );
     }
+
+    struct EnvVarGuard {
+        key: String,
+        original_value: Option<String>,
+    }
+
+    impl EnvVarGuard {
+        fn new(key: &str, new_value: &str) -> Self {
+            let original_value = std::env::var(key).ok();
+            std::env::set_var(key, new_value);
+            EnvVarGuard {
+                key: key.to_string(),
+                original_value,
+            }
+        }
+    }
+
+    impl Drop for EnvVarGuard {
+        fn drop(&mut self) {
+            if let Some(ref value) = self.original_value {
+                std::env::set_var(&self.key, value);
+            } else {
+                std::env::remove_var(&self.key);
+            }
+        }
+    }
```

Comment:
Strange.  I thought we had something like this in the code base already but I cannot find it now.